contracts/PercentageTransferManager/src/lib.rs

```diff
@@ -0,0 +1,593 @@
+        ///
+        /// # Arguments
+        /// * `identity` - Identity of the token holder whose exemption status needs to change
+        /// * `is_exempted` - New exemption status of the identity
+        #[ink(message)]
+        fn modify_exemption_list(&mut self, identity: IdentityId, is_exempted: bool) {
+            assert!(self.env().caller() == *self.owner.get(), "Not Authorized");
+            assert!(
+                self._is_exempted_or_not(&identity) != is_exempted,
+                "Must change setting"
+            );
+            self._modify_exemption_list(identity, is_exempted);
+        }
+
+        /// To exempt the given Identities from the restriction
+        ///
+        /// # Arguments
+        /// * `identities` - Identities of the token holders whose exemption status needs to change
+        /// * `exemptions` - New exemption status of the identities
+        #[ink(message)]
+        fn modify_exemption_list_batch(
+            &mut self,
+            identities: Vec<IdentityId>,
+            exemptions: Vec<bool>,
+        ) {
+            assert!(
+                identities.len() == exemptions.len(),
+                "Arguments length mismatch"
+            );
+            for i in 0..identities.len() {
+                self._modify_exemption_list(identities[i], exemptions[i]);
+            }
+        }
+
+        /// Transfer ownership of the smart extension
+        ///
+        /// # Arguments
+        /// * `new_owner` - AccountId of the new owner
+        #[ink(message)]
+        fn transfer_ownership(&mut self, new_owner: AccountId) {
+            assert!(self.env().caller() == *self.owner.get(), "Not Authorized");
+            self.env().emit_event(TransferOwnership {
+                old_owner: self.env().caller(),
+                new_owner: new_owner,
+            });
+            self.owner.set(new_owner);
+        }
+
+        /// Simply returns the current value of `max_allowed_percentage`.
+        #[ink(message)]
+        fn get_max_allowed_percentage(&self) -> u128 {
+            *self.max_allowed_percentage.get()
+        }
+
+        /// Simply returns the current value of `allow_primary_issuance`.
+        #[ink(message)]
+        fn is_primary_issuance_allowed(&self) -> bool {
+            *self.allow_primary_issuance.get()
+        }
+
+        /// Simply returns the current value of `owner`.
+        #[ink(message)]
+        fn owner(&self) -> AccountId {
+            *self.owner.get()
+        }
+
+        /// Function to know whether given Identity is exempted or not
+        #[ink(message)]
+        fn is_exempted_or_not(&self, of: IdentityId) -> bool {
+            *self.exemption_list.get(&of).unwrap_or(&false)
+        }
+
+        fn _is_exempted_or_not(&self, of: &IdentityId) -> bool {
+            *self.exemption_list.get(of).unwrap_or(&false)
+        }
+
+        fn _modify_exemption_list(&mut self, identity: IdentityId, is_exempted: bool) {
```

Comment:
Just making sure, the outside world can not call this, right? I guess only functions with attribute `#[ink(message)]` can be called by the outside world?

Reply:
Yeah, It will not be allowed to call from outside.